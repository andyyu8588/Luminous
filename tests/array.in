class Object1 {
    public x;
    public constructor() {
        this.x = "pogg";
    }
    public printPogg() {
        print(this.x);
    }
}

class Object2 inherits Object1 {
    public x;
    public constructor() {
        this.x = "Not pogg";
    }
    public printPogg() {
        print(this.x);
    }
}

class Container {
    public arr;
    public constructor() {
        this.arr = [Object1(), Object2()];
    }
    public printContainer() {
        print(this.arr);
        for(i from 0 to 2 by 1) {
            this.arr[i].printPogg();
        }
    }
}

container = Container();
container.printContainer();
globalNum = 3.1414;
globalStr = "this is a string";

globalArr = [container, globalNum, globalStr];
print(globalArr);

changeArrVar = null;

function printRandom() {
    randomArr = [1, 2, 3, 4, 5];
    function changeArr() {
        randomArr[0] = "ok";
        randomArr[2] = "nice!";
        randomArr[4] = "Lets gooo!";
    }
    changeArrVar = changeArr;
    return randomArr;
}

printRandomReturnVal = printRandom();
changeArrVar();
print(printRandomReturnVal);

function modifyArray(arr) {
    arr[0] = "modified";
    arr[1] = "corrupted";
}

a1 = [1, 2];
modifyArray(a1);
print(a1);
print(a1[1 + 0]);
