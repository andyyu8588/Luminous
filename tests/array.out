[Object1 instance, Object2 instance]
pogg
Not pogg
[Container instance, 3.1414, this is a string]
[ok, 2, nice!, 4, Lets gooo!]
[modified, corrupted]
corrupted
